cosmwasm/ibc-union/core/light-client-interface/src/lib.rs

```diff
@@ -170,6 +186,71 @@ fn client_impl<T: IbcClient>(
     Ok(addr)
 }
 
+pub struct StateUpdate<T: IbcClient> {
+    pub height: u64,
+    pub client_state: Option<T::ClientState>,
+    pub consensus_state: T::ConsensusState,
+    pub storage_writes: StorageWrites,
+}
+
+impl<T: IbcClient> StateUpdate<T> {
+    pub fn new(height: u64, consensus_state: T::ConsensusState) -> Self {
+        StateUpdate {
+            height,
+            consensus_state,
+            client_state: None,
+            storage_writes: Default::default(),
+        }
+    }
+
+    pub fn overwrite_client_state(mut self, client_state: T::ClientState) -> Self {
+        self.client_state = Some(client_state);
+        self
+    }
+
+    pub fn add_storage_write<V: Encode<T::Encoding>>(mut self, key: Bytes, value: V) -> Self {
+        self.storage_writes.insert(key, value.encode().into());
+        self
+    }
+}
+
+/// Client creation output type
+pub struct ClientCreation<T: IbcClient> {
+    /// The client state that is going to be stored by IBC. If set to `None`, IBC will store the
+    /// client state given by the creator as is
+    pub client_state: Option<T::ClientState>,
+    /// Custom events that will be emitted by IBC.
+    pub events: Vec<VerifyCreationResponseEvent>,
+    /// Arbitrary storage saves to the client's corresponding storage. These are accessible to the
+    /// client at any time.
+    pub storage_writes: StorageWrites,
+}
+
+impl<T: IbcClient> ClientCreation<T> {
+    pub fn empty() -> Self {
+        Self {
+            client_state: None,
+            events: Vec::new(),
+            storage_writes: Default::default(),
+        }
+    }
```

Comment:
minor nit: change this method name to `new`, and rename `ClientCreation` to `ClientCreationResult`

Reply:
promise bro, just one more nit bro